### PR TITLE
`Programming exercises`: Improve appearance of results and task bar

### PR DESCRIPTION
**Pull request by JohannesStoehr**
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).
#### Client
- [x] I followed the [coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/) and ensured that the layout is responsive.
- [x] I added multiple screenshots/screencasts of my UI changes.

### Motivation and Context

Make the Ui of programming exercises a bit more intuitive, so students don't mix up the different elements and there is a more clearer distinction.

### Description

The "show all"/"hide all results" button now has a triangle like other show/hide toggles in Artemis, making it more intuitive. Additionally it always stays at the same place and is not put at the very bottom when showing all results.
The task bar showing icons for all tasks now uses the normal Tooltips and the horizontal line through all tasks is replaced by a small title before all tasks to make it clearer what these icons are representing.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Student
- 1 Programming Exercise

1. Log in to Artemis
2. Go to the exercise
3. Look at the UI elements before submitting once and check for their understandability
4. Submit once
5. Look at the UI elements and check for their understandability
6. Submit another time, so there are at least two results
7. Look at the UI elements one last time and check for their understandability

#### Code Review
- [x] Review 1
- [x] Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### Screenshots

Before:
Bildschirm­foto 2023-01-12 um 17 08 27
Now:
Bildschirm­foto 2023-01-12 um 17 05 28

### CODE REVIEW

**Review by PhilipWerz — APPROVED** [review]
Everything looks and works as expected.  Tested on TS1.

Note:
No "Recent Results" View after first commit.

**Review by timo07k — COMMENTED** [review]
It generally works as expected, but I am not sure if it is supposed to look like this after the first submission:

image

For me it looks a bit inconsistent with how it is displayed after the second submission:

image

Why does it not already look like this after the first submission?

Tested on TS1 and TS2.

File: src/main/webapp/i18n/en/programmingExercise.json [hunk 1]
```diff
@@ -125,7 +125,8 @@
             "problemStatement": {
                 "title": "Problem Statement",
                 "taskCommand": "Task",
-                "testCaseCommand": "Insert Test Case"
+                "testCaseCommand": "Insert Test Case",
+                "tasksOfExercise": "Tasks of exercise:"
```
Comment (ge65cer) [inline src/main/webapp/i18n/en/programmingExercise.json:129]:
```suggestion
                "tasksOfExercise": "Exercise tasks:"
```

**Review by LiaLism — DISMISSED** [review]
The results and the taskbar match the screenshot. Tested on ts1.

File: src/main/webapp/i18n/en/programmingExercise.json (hunk 1 above)
Comment (Strohgelaender) [reply src/main/webapp/i18n/en/programmingExercise.json:129]: Maybe even just "Tasks"? There is only one exercise open.

**Review by ge65cer — APPROVED** [review]

**Review by LiaLism — APPROVED** [review]
Everything looks as expected. Tested on ts1.

**Comment by ge65cer** [comment]
Related issue: https://github.com/ls1intum/Artemis/issues/1070

**Pull request merged**